docs/general/ddl/data-types.rst

```diff
@@ -303,1283 +458,2517 @@ Casting from ``OBJECT`` to ``JSON``::
+
+
+.. _data-types-container:
+
+Container types
+===============
+
+Container types are types with :ref:`nonscalar <gloss-nonscalar>` values that
+may contain other values:
+
+.. contents::
+   :local:
+   :depth: 3
+
+
+.. _data-types-objects:
+
+Objects
+-------
+
+
+.. _type-object:
+
+``OBJECT``
+''''''''''
+
+An object is structured as a collection of key-values.
+
+An object can contain any other type, including further child objects. An
+``OBJECT`` column can be schemaless or can have a defined (i.e., enforced)
+schema.
+
+Objects are not the same as JSON objects, although they share a lot of
+similarities. However, objects can be :ref:`inserted as JSON strings
+<data-types-object-json>`.
+
+Syntax::
+
+    <columnName> OBJECT
+        [ ({DYNAMIC|STRICT|IGNORED}) ]
+        [ AS ( <columnDefinition>* ) ]
+
+The only required syntax is ``OBJECT``.
+
+The column policy (``DYNAMIC``, ``STRICT``, or ``IGNORED``) is optional and
+defaults to :ref:`DYNAMIC <type-object-columns-dynamic>`.
+
+If the optional list of subcolumns (``columnDefinition``) is omitted, the
+object will have no schema. CrateDB will create a schema for :ref:`DYNAMIC
+<type-object-columns-dynamic>` objects upon first insert.
+
+Example::
+
+    cr> CREATE TABLE my_table (
+    ...     title TEXT,
+    ...     quotation OBJECT,
+    ...     protagonist OBJECT(STRICT) AS (
+    ...         age INTEGER,
+    ...         first_name TEXT,
+    ...         details OBJECT AS (
+    ...             birthday TIMESTAMP WITH TIME ZONE
+    ...         )
+    ...     )
+    ... );
+    CREATE OK, 1 row affected (... sec)
+
+::
+
+    cr> INSERT INTO my_table (
+    ...     title,
+    ...     quotation,
+    ...     protagonist
+    ... ) VALUES (
+    ...     'Alice in Wonderland',
+    ...     {
+    ...         "words" = 'Curiouser and curiouser!',
+    ...         "length" = 3
+    ...     },
+    ...     {
+    ...         "age" = '10',
+    ...         "first_name" = 'Alice',
+    ...         "details" = {
+    ...             "birthday" = '1852-05-04T00:00Z'::TIMESTAMPTZ
+    ...         }
+    ...     }
+    ... );
+    CREATE OK, 1 row affected (... sec)
+
+.. HIDE:
+
+    cr> REFRESH TABLE my_table;
+    REFRESH OK, 1 row affected (... sec)
+
+::
+
+    cr> SELECT
+    ...     protagonist['first_name'] AS name,
+    ...     date_format(
+    ...         '%D %b %Y',
+    ...         'GMT',
+    ...         protagonist['details']['birthday']
+    ...      ) AS born,
+    ...     protagonist['age'] AS age
+    ... FROM my_table;
+    +-------+--------------+-----+
+    | name  | born         | age |
+    +-------+--------------+-----+
+    | Alice | 4th May 1852 |  10 |
+    +-------+--------------+-----+
+    SELECT 1 row in set (... sec)
+
+.. HIDE:
+
+    cr> DROP TABLE my_table;
+    DROP OK, 1 row affected (... sec)
+
+
+.. _type-object-column-policy:
+
+Object column policy
+....................
+
+
+.. _type-object-columns-strict:
+
+``STRICT``
+``````````
```

Comment:
This section level looks not so nice when rendered in HTML, it is very small. Maybe we've never used such deep-levels before? Think this requires style tuning to keep it nice and readable.

Reply:
I believe https://github.com/crate/crate-docs-theme/commit/f8692c9 might already have addressed this. I will check the outcome and mint a new release if that helps.

Reply:
Indeed, @norosa's patch improves the situation -- thanks! Before/after:

<img src="https://user-images.githubusercontent.com/453543/128542603-82522fed-ea04-4193-bcde-2d7496dcce56.png" width="350">

<img src="https://user-images.githubusercontent.com/453543/128542617-d916b842-747a-4a34-824d-3424f3eabd8a.png" width="350">

Reply:
It looks much better now, see [1].

[1] https://crate--11631.org.readthedocs.build/en/11631/general/ddl/data-types.html#object-column-policy

Reply:
Great, thanks!